subprojects/core/src/test/groovy/org/gradle/initialization/DefaultGradleLauncherFactoryTest.groovy

```diff
@@ -90,21 +92,33 @@ class DefaultGradleLauncherFactoryTest extends Specification {
         launcher.gradle.services.get(BuildEventConsumer) == eventConsumer
     }
 
-    def "marks BuildScanRequest as requested when build scan startparameter is set"() {
+    def "initializes build scan config"() {
         given:
         startParameter.setBuildScan(true)
+
         when:
         def launcher = factory.newInstance(startParameter, Stub(BuildRequestContext), sessionServices)
+        def c = buildScanConfig(launcher)
+
         then:
-        launcher.gradle.getServices().get(BuildScanRequest).collectRequested()
+        c.enabled
+        !c.disabled
     }
 
     def "marks BuildScanRequest as disabled when no build scan startparameter is set"() {
         given:
         startParameter.setNoBuildScan(true)
+
         when:
         def launcher = factory.newInstance(startParameter, Stub(BuildRequestContext), sessionServices)
+        def c = buildScanConfig(launcher)
+
         then:
-        launcher.gradle.getServices().get(BuildScanRequest).collectDisabled()
+        !c.enabled
+        c.disabled
+    }
```

Comment:
Should we also have a test for the case where nothing is specified explicitly? i.e. neither setBuildScan nor setNoBuildScan